packages/blade/src/components/Button/BaseButton/BaseButton.stories.tsx

```diff
@@ -51,14 +55,52 @@ const BaseButtonTemplate: ComponentStory<typeof BaseButtonComponent> = ({
   ...args
 }) => {
   const IconComponent = iconMap[(icon as unknown) as string];
-
   return (
     <BaseButtonComponent icon={IconComponent} {...args}>
       {children}
     </BaseButtonComponent>
   );
 };
 
+const ButtonLoadingExample = (args: any): React.ReactElement => {
+  const [loading, setLoading] = useState(false);
+
+  return (
+    <>
+      <BaseButtonComponent onClick={(): void => setLoading(!loading)}>
```

Comment:
It's there

> Also a suggestion(can ignore): we can use checkbox to toggle this(just like a switch)

I'll do that when we have the Checkbox component otherwise the native html input currently doesn't work on react-native.

Reply:
Discussed, Move the toggle CTA below the button, make it secondary xsmall